### PR TITLE
Add TruncatedSVD

### LINKED ISSUE #49: Standard and Truncated SVD
Standard SVD on GPU is in place without API, but holding until can see how to do truncated SVD.  Also, seems sklearn truncated SVD uses lots of memory despite being truncated -- can one optimize more for memory vs. time?

### PR DESCRIPTION
**Pull request by navdeep-G**
Resolves #49 
Utilizes `cusolverDnSsyevd` to compute eigenvalues and eigenvectors, which can be used to get singular values, the U matrix, etc.

Some plots to display reconstruction error and speed benchmarks (conducted on random data):
tsvd_k100

tsvd_k49_n50

tsvd_bench

### CODE REVIEW

File: src/gpu/tsvd/tsvd.cu [hunk 1]
```diff
@@ -12,6 +12,14 @@
...
 {
 
+/**
+ * Division utility to get explained variance ratio
+ *
+ * @param XVar
+ * @param XVarSum
+ * @param ExplainedVarRatio
+ * @param context
+ */
```
Comment (RAMitchell) [inline src/gpu/tsvd/tsvd.cu]: Possible divide by zero in this function?

**Comment by mdymczyk** [comment]
Will try this out myself later in the day but looks really good 👍  We should strive to make the whole repo like this.

File: src/gpu/tsvd/tsvd.cu (hunk 1 above)
Comment (navdeep-G) [reply src/gpu/tsvd/tsvd.cu]: ah, good point. will fix.

**Comment by navdeep-G** [comment]
Thank you! @mdymczyk @RAMitchell

**Comment by navdeep-G** [comment]
@RAMitchell Right now the python tests run through pytest, but no consistent way of writing them. And unifying matrix utilities would be great. I would love to have a single data representation instead of several across algos.

**Comment by mdymczyk** [comment]
@navdeep-G @RAMitchell one more question - from what I see this is a single GPU implementation, right? Is it possible to distribute SVM across GPUs (either for speed or when matrices are too big)?

**Comment by RAMitchell** [comment]
Possibly could do something with cublasXT.  Have never used it before.

**Comment by pseudotensor** [comment]
It sucks major.  Very slow

**Pull request merged**